src/main/java/zmq/ZMQ.java

```diff
@@ -822,14 +822,23 @@ public Selector get()
             return selector;
         }
 
+        void close()
+        {
+            if (selector != null) {
+                try {
+                    selector.close();
+                    selector = null;
+                }
+                catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
         @Override
         public void finalize()
         {
```

Comment:
I agree, there are a lot of ramifications involving the use of `finalize` that most people don't understand. I would go so far as to say that you should never override it unless you know exactly what your doing.